### DISCUSSION TITLE
`RuleSetCompiler.js` fails with TypeError

### DISCUSSION THREAD

**Discussion by vvanpo** [webpack/webpack#12454] [Q&A]
I upgraded from Webpack 4 to 5, and am now seeing:

```
foo/node_modules/webpack/lib/rules/RuleSetCompiler.js:228
                                fn: str => str.startsWith(condition)
                                               ^

TypeError: Cannot read property 'startsWith' of null
    at Object.fn (foo/node_modules/webpack/lib/rules/RuleSetCompiler.js:228:20)
    at execRule (foo/node_modules/webpack/lib/rules/RuleSetCompiler.js:100:22)
    at execRule (foo/node_modules/webpack/lib/rules/RuleSetCompiler.js:120:6)
    at Object.exec (foo/node_modules/webpack/lib/rules/RuleSetCompiler.js:139:6)
    at foo/node_modules/webpack/lib/NormalModuleFactory.js:433:34
    at foo/node_modules/webpack/lib/NormalModuleFactory.js:118:11
    at foo/node_modules/webpack/lib/NormalModuleFactory.js:626:8
    at foo/node_modules/webpack/lib/NormalModuleFactory.js:762:5
    at finishResolved (foo/node_modules/webpack/node_modules/enhanced-resolve/lib/Resolver.js:278:11)
    at foo/node_modules/webpack/node_modules/enhanced-resolve/lib/Resolver.js:342:25
    at foo/node_modules/webpack/node_modules/enhanced-resolve/lib/Resolver.js:409:24
    at eval (eval at create (foo/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:10:1)
    at foo/node_modules/webpack/node_modules/enhanced-resolve/lib/Resolver.js:409:24
    at eval (eval at create (foo/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:25:1)
    at foo/node_modules/webpack/node_modules/enhanced-resolve/lib/DescriptionFilePlugin.js:88:10
    at foo/node_modules/webpack/node_modules/enhanced-resolve/lib/Resolver.js:409:24
```

The offending code is here: https://github.com/webpack/webpack/blob/v5.15.0/lib/rules/RuleSetCompiler.js#L228

Is this indicative of a module rule that is misconfigured? I'm not sure how to interpret the offending function, I assume it is supposed to be passed a [condition input](https://webpack.js.org/configuration/module/#rule-conditions), i.e. a resource or issuer string, so how could those ever be `null`?

**Comment by alexander-akait** [comment]
Why you check types of webpack inside own project?

**Reply by vvanpo** [reply]
I'm not sure I understand the question. The above error occurs when I attempt to run Webpack.

**Comment by vankop** [comment]
Could you provide your webpack config please?

**Reply by vvanpo** [reply]
I unfortunately can't provide my entire Webpack config, but I have been able to narrow it down to a single rule:

```js
{
    issuer: `${process.cwd()}/foo.js`,
},
```

It doesn't matter what other conditions I add to this rule, or what loaders I add to the `use` option; the `webpack/lib/rules/RuleSetCompiler.js:228 ... fn: str => str.startsWith(condition)` error persists.

**Reply by vvanpo** [reply]
Interestingly, if I wrap the string in `new RegExp(...)`, there is no error. So it appears that Webpack 5 has a regression where `issuer` conditions can not be strings.

**Reply by alexander-akait** [reply]
Maybe you can create simple example?
